src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -73,167 +74,96 @@ public override void Accept(OperationVisitor visitor)
-
-        private static IArgument DeriveArgument(int parameterIndex, int argumentIndex, ImmutableArray<BoundExpression> boundArguments, ImmutableArray<string> argumentNamesOpt, ImmutableArray<RefKind> argumentRefKindsOpt, ImmutableArray<Symbols.ParameterSymbol> parameters, SyntaxNode invocationSyntax)
-        {
-            if ((uint)argumentIndex >= (uint)boundArguments.Length)
-            {
-                // Check for an omitted argument that becomes an empty params array.
-                if (parameters.Length > 0)
+            return (ImmutableArray<IArgument>) s_callToArgumentsMappings.GetValue(
+                boundNode, 
+                (n) =>
                 {
-                    Symbols.ParameterSymbol lastParameter = parameters[parameters.Length - 1];
-                    if (lastParameter.IsParams)
-                    {
-                        return new Argument(ArgumentKind.ParamArray, lastParameter, CreateParamArray(lastParameter, boundArguments, argumentIndex, invocationSyntax));
-                    }
-                }
-
-                // There is no supplied argument and there is no params parameter. Any action is suspect at this point.
-                return new SimpleArgument(null, new InvalidExpression(invocationSyntax, ImmutableArray<IOperation>.Empty));
-            }
-
-            return s_argumentMappings.GetValue(
-                boundArguments[argumentIndex],
-                (argument) =>
-                {
-                    string nameOpt = !argumentNamesOpt.IsDefaultOrEmpty ? argumentNamesOpt[argumentIndex] : null;
-                    Symbols.ParameterSymbol parameterOpt = (uint)parameterIndex < (uint)parameters.Length ? parameters[parameterIndex] : null;
-
-                    if ((object)nameOpt == null)
+                    //TODO: https://github.com/dotnet/roslyn/issues/18722
+                    //      Right now, for errornous code, we try to recover as much data as we can
+                    //      and expose them as IArguments, so user needs to check IsInvalid firdt before using 
+                    //      anything we returned. Need to implement a new interface for invalid invocation instead.
+                    if (n.HasErrors)
                     {
-                        RefKind refMode = argumentRefKindsOpt.IsDefaultOrEmpty ? RefKind.None : argumentRefKindsOpt[argumentIndex];
-
-                        if (refMode != RefKind.None)
+                        ArrayBuilder<IArgument> argumentsWithErrors = ArrayBuilder<IArgument>.GetInstance(boundArguments.Length);
+                        for (int a = 0; a < boundArguments.Length; a++)
                         {
-                            return new Argument(ArgumentKind.Explicit, parameterOpt, argument);
+                            var parameter = argumentsToParametersOpt.IsDefault
```

Comment:
>parameter [](start = 32, length = 9)

We probably shouldn't tie arguments to parameters in this case, we might get in a situation when multiple arguments are linked to the same parameter.